scheduler/workflow_test.go

```diff
@@ -121,3 +125,172 @@ func TestCollectPublishWorkflow(t *testing.T) {
 		})
 	})
 }
+
+// The mocks below are here for testing work submission
+type Mock1 struct {
+	sync.Mutex
+	count      int
+	errorIndex int
+	delay      time.Duration
+	queue      map[string]int
+}
+
+func (m *Mock1) CollectMetrics([]core.Metric, time.Time, string) ([]core.Metric, []error) {
+	return nil, nil
+}
+
+func (m *Mock1) Work(j job) queuedJob {
+	m.Lock()
+	defer m.Unlock()
+	time.Sleep(m.delay)
+	m.queue[j.TypeString()]++
+	return m
+}
+
+func (m *Mock1) Promise() promise.Promise {
+	return m
```

Comment:
I didn't want to test the `Promise` behavior in this test so I mocked completely. I figured the exported `Promise` functions are pretty fixed.